### DISCUSSION TITLE
Is any way to get NUMBER column's  precision and scale

### DISCUSSION THREAD

**Discussion by zeales-go** [sijms/go-ora#93] [Q&A] [answered]
use  github.com/jmoiron/sqlx

mdb, _ = sqlx.Open("oracle", ...)
que, _ = mdb.Queryx("select x,y,z,optime from ttt")
oraTypes, _ := que.ColumnTypes()
for i := range oraTypes {
	p, s, ok := oraTypes[i].DecimalSize()
	fmt.Println(oraTypes[i].DatabaseTypeName(), p, s, ok)
}

NUMBER 0 0 false
NUMBER 0 0 false
NCHAR 0 0 false
DATE 0 0 false

CREATE TABLE ttt (
  x                              NUMBER               NULL,
  y                              NUMBER(12,2)         NULL,
  z                              VARCHAR2(200)        NULL,
  optime                         DATE                 NULL
);

how can i know the exactly column type of X  or Y.

**Comment by sijms** [comment, marked as answer]
you can know that by 2 ways
1- from go_ora.DataSet.Cols array where you can find column meta-data 
2- form oracle database by query ALL_TAB_COLUMNS
